rzup/src/lib.rs

```diff
@@ -11,64 +11,378 @@
-            if let Some(code) = output.status.code() {
-                eprintln!("Exit code: {}", code);
-            }
+    /// Creates a new Rzup instance using default environment paths.
+    ///
+    /// This will initialize the registry, settings, and scan the environment for installed components.
+    pub fn new() -> Result<Self> {
+        let environment = Environment::new()?;
+        let mut registry = Registry::new(&environment)?;
+        Self::initialize_settings(&environment, &mut registry)?;
+        registry.scan_environment(&environment)?;
+
+        Ok(Self {
+            environment,
+            registry,
+        })
+    }
+
+    /// Creates a new Rzup instance with a custom root directory.
+    ///
+    /// # Arguments
+    /// * `root` - The root directory path for storing components and settings
+    pub fn with_root<P: Into<PathBuf>>(root: P) -> Result<Self> {
+        let environment = Environment::with_root(root)?;
+        let mut registry = Registry::new(&environment)?;
+        Self::initialize_settings(&environment, &mut registry)?;
+        registry.scan_environment(&environment)?;
+
+        Ok(Self {
+            environment,
+            registry,
+        })
+    }
+
+    fn initialize_settings(environment: &Environment, registry: &mut Registry) -> Result<()> {
+        if !environment.settings_path().exists() {
+            environment.emit(RzupEvent::Debug {
+                message: format!(
+                    "Creating new settings file at {}",
+                    environment.settings_path().display()
+                ),
+            });
         }
+        registry.settings().save(environment)?;
+        Ok(())
+    }
 
+    /// Sets an event handler for receiving notifications about operations.
+    ///
+    /// # Arguments
+    /// * `handler` - Function that will be called for each event
+    pub fn set_event_handler<F>(&mut self, handler: F)
+    where
+        F: Fn(RzupEvent) + Send + Sync + 'static,
+    {
+        self.environment.set_event_handler(handler);
+    }
+
+    /// Installs all default components.
+    ///
+    /// # Arguments
+    /// * `force` - If true, reinstalls even if already installed
+    pub fn install_all(&mut self, force: bool) -> Result<()> {
+        self.registry
+            .install_all_components(&self.environment, force)?;
+        self.registry.scan_environment(&self.environment)?;
         Ok(())
     }
 
-    pub fn uninstall() -> Result<()> {
-        let cargo_bin_path = dirs::home_dir()
-            .ok_or_else(|| anyhow!("Could not determine home directory"))?
-            .join(".cargo/bin");
+    /// Installs a specific component version.
+    ///
+    /// # Arguments
+    /// * `component` - Component identifier (e.g. "rust", "cargo-risczero")
+    /// * `version` - Specific version to install, or None for latest
+    /// * `force` - If true, reinstalls even if already installed
+    pub fn install_component(
+        &mut self,
+        component: &str,
```

Comment:
Have you considered making this be an `enum` instead of a string? Both here and in other places in the library (the HashMaps could be EnumMap or otherwise). At least from an API standpoint it might make a little nicer to use (easier to discover the valid components, and let you detect when you have chosen an invalid component at build time vs run time).